api/src/pcapi/core/finance/api.py

```diff
@@ -1333,6 +1333,41 @@ def cancel_pricing(
     return pricing
 
 
+def update_finance_event_pricing_date(
+    stock: offers_models.Stock | educational_models.CollectiveStock,
+    new_pricing_ordering_date: datetime.datetime,
+) -> None:
+    pricing_point_id = stock.offer.venue.current_pricing_point_id
+    finance_events_from_pricing_point = models.FinanceEvent.query.options(
+        sqla.orm.joinedload(models.FinanceEvent.pricings)
+    ).filter(models.FinanceEvent.pricingPointId == pricing_point_id)
+
+    bookings_of_this_stock = bookings_models.Booking.query.with_entities(bookings_models.Booking.id).filter(
+        bookings_models.Booking.stockId == stock.id
+    )
+    finance_events_from_stock = finance_events_from_pricing_point.filter(
+        models.FinanceEvent.bookingId.in_(bookings_of_this_stock)
+    ).all()
+
+    if finance_events_from_stock:
+        old_pricing_ordering_date = finance_events_from_stock[0].pricingOrderingDate
+        for finance_event in finance_events_from_stock:
+            finance_event.pricingOrderingDate = new_pricing_ordering_date
```

Comment:
[bug] Ici on indique que la nouvelle `pricingOrderingDate` est la nouvelle date du stock. En réalité, ça n'est pas tout à fait ça : `pricingOrderingDate` est calculé à partir de la date du stock, mais aussi de la date de validation de la réservation (et aussi de la date de liaison entre le lieu et son point de valorisation). Il faudrait donc ici refaire ce calcul : 

```suggestion
            finance_event.pricingOrderingDate = _get_pricing_ordering_date(finance_event.booking or finance_event.collectiveBooking)
```

Attention : pour cela, il faut que la date du stock ait bien été modifiée avant que l'on appelle `update_finance_event_pricing_date()`. C'est le cas dans ta PR, sauf dans `edit_stock()` où il faudrait déplacer l'appel à `update_finance_event_pricing_date()` après la mise à jour de `stock.beginningDatetime`.

Désolé, j'avais manqué ça dans mes précédentes revues.